src/modules/blockchainApplication/manage/store/reducer.js

```diff
@@ -24,7 +35,34 @@ export const pins = (state = [], { type, chainId }) => {
 /**
  *
  * @param {Object} state
- * @param {type: String, encryptedAccount: Object} action
+ * @param {type: String, data: Object} action
+ */
+export const applications = (state = initialState.applications, { type, data }) => {
+  switch (type) {
+    case actionTypes.addApplicationByChainId:
+      // In cases where a new node for an existing application is being added,
+      // the new node url should be appended to the nodeURL array of the application
+      if (data.chainID in state) {
+        state[data.chainID].nodeURL.push(data.nodeURL);
+      } else {
+        state[data.chainID] = data;
+      }
+      return state;
+
+    case actionTypes.deleteApplicationByChainId: {
+      delete state[data];
+      return { ...state };
+    }
```

Comment:
improve naming please data is a bit wired for me
```suggestion
export const applications = (state = initialState.applications, { type, application, chainId }) => {
  switch (type) {
    case actionTypes.addApplicationByChainId:
      // In cases where a new node for an existing application is being added,
      // the new node url should be appended to the nodeURL array of the application
      if (application.chainID in state) {
        state[application.chainID].nodeURL.push(data.nodeURL);
      } else {
        state[application.chainID] = data;
      }
      return state;

    case actionTypes.deleteApplicationByChainId: {
      delete state[chainId];
      return { ...state };
    }
```